### PR TITLE
react-select: update props to match Select and Async source files

### PR DESCRIPTION
**Pull request by lediur**
I went through the `react-select` repo and updated the props to match what's currently in Select.js and Async.js. I'll go through and add line comments for each added / removed prop.

### CODE REVIEW

File: react-select/react-select.d.ts [hunk 1]
```diff
@@ -19,48 +19,75 @@ declare namespace ReactSelect {
...
+         */
+        selectValue: (option: Option) => void;
+
+        /**
+         * Array of currently selected options.
+         */
+        valueArray: Option[];
+    }
+
     interface ReactSelectProps extends __React.Props<ReactSelect> {
```
Comment (lediur) [inline react-select/react-select.d.ts]: For the below, refer to https://github.com/JedWatson/react-select/blob/master/src/Select.js#L29 and its contents

File: react-select/react-select.d.ts [hunk 2]
```diff
@@ -19,48 +19,75 @@ declare namespace ReactSelect {
         clearableValue?: boolean;
     }
 
+    interface MenuRendererProps {
```
Comment (lediur) [inline react-select/react-select.d.ts]: See https://github.com/JedWatson/react-select/#effeciently-rendering-large-lists-with-windowing for a list of `menuRenderer` props

**Comment by vvakame** [comment]
We can't get response from authors.
please ping me again in 2 days. I'll merge this PR.
and, please resolve conflict.

**Comment by lediur** [comment]
@vvakame Pinging you as requested. Thanks for your help!

**Pull request merged**